marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/ExecEndpointImpl.java

```diff
@@ -103,7 +115,38 @@ private ExecEndpointImpl getEndpoint() {
         @Override
         public void awaitCompletion() {
             setPhase(WorkPhase.RUNNING);
-            logger.trace("exec running endpoint={} work={}", getEndpointPath(), callContext.getWorkUnit());
+            logger.trace("exec running endpoint={} work={}", getEndpointPath(), getCallContext().getWorkUnit());
+
+            if(threadCount == 1)
+                processOutput(getCallContext());
```

Comment:
After we handle 

*  the simple case of a single call context and single thread
*  the case of multiple call contexts and multiple threads

We'll also need to deal with the case where there are multiple call contexts and a single thread.